### PR TITLE
simplify host fs mounting

### PR DESCRIPTION
**Pull request by dustin-bingham**

### CODE REVIEW

**Review by samiura — APPROVED** [review]

File: templates/kernel-collector-daemonset.yaml [hunk 1]
```diff
@@ -124,34 +124,25 @@ spec:
...
-            readOnly: true
 {{- if (default false .Values.kernelCollector.useDockerMetadata) }}
           - mountPath: /var/run/docker.sock
             name: docker-sock
             readOnly: false
 {{- end }}
 {{- if hasKey .Values "debug" }}
 {{- if (default false .Values.debug.storeMinidump) }}
-          - mountPath: /hostfs/run/network-explorer
+          - mountPath: /hostfs/var/run/network-explorer
```
Comment (bjandras) [inline templates/kernel-collector-daemonset.yaml:145]: But the `EBPF_NET_MINIDUMP_DIR` environment variable is being set to `/hostfs/run/network-explorer/minidump`.

File: templates/kernel-collector-daemonset.yaml [hunk 2]
```diff
@@ -124,34 +124,25 @@ spec:
...
-            name: lib-modules
+          - mountPath: /hostfs/
+            name: host-root
             readOnly: true
 {{- if (default true .Values.kernelCollector.fetchKernelHeaders) }}
-          - mountPath: /hostfs/etc
-            name: etc
-            readOnly: true
-          - mountPath: /hostfs/cache
+          - mountPath: /hostfs/var/cache
```
Comment (bjandras) [inline templates/kernel-collector-daemonset.yaml:131]: If the host's entire filesystem under mounted to `/hostfs`, doesn't that also include `/var/cache` as `/hostfs/var/cache`?

Comment (dustin-bingham) [reply templates/kernel-collector-daemonset.yaml:131]: yeah, but mounted RO.  these mounts are editable.

File: templates/kernel-collector-daemonset.yaml (hunk 1 above)
Comment (dustin-bingham) [reply templates/kernel-collector-daemonset.yaml:145]: Fixed

File: templates/kernel-collector-daemonset.yaml [hunk 3]
```diff
@@ -105,7 +105,7 @@ spec:
 {{- if hasKey .Values "debug" }}
 {{- if (default false .Values.debug.storeMinidump) }}
             - name: "EBPF_NET_MINIDUMP_DIR"
-              value: "/hostfs/run/network-explorer/minidump"
+              value: "/hostfs/var/run/network-explorer/minidump"
```
Comment (bjandras) [inline templates/kernel-collector-daemonset.yaml:108]: Other deployments (cloud-collector, reducer, ...) mount at `/hostfs/run/network-explorer/minidump`. Is this change necessary?

Comment (dustin-bingham) [reply templates/kernel-collector-daemonset.yaml:108]: I believe so - we won't be able to write to /hostfs/run.  I can update the other deploys to make them consistent?

Comment (bjandras) [reply templates/kernel-collector-daemonset.yaml:108]: Up to you.

Comment (dustin-bingham) [reply templates/kernel-collector-daemonset.yaml:108]: Yeah lemme add them too.

Comment (dustin-bingham) [reply templates/kernel-collector-daemonset.yaml:108]: Done.

Comment (bjandras) [reply templates/kernel-collector-daemonset.yaml:108]: 👍
